### DISCUSSION TITLE
AppleScript to run PyDPainter

### DISCUSSION THREAD

**Discussion by PerspexSphinx** [mriale/PyDPainter#125] [Show and tell]
Hi, PyDPainter is really **GREAT**…
I use it on my mac,  so I thought it would be nice to run it with just a  double mouse click, so I wrote a simple AppleScript too run it…  

```
(* My Brilliant applescript code
to run PyDPainter
version alpha 1  
Note: change path to your location of python3  
and your location & version of PyDPainter *)  

on run  

	activate  
	
	do shell script "/usr/local/bin/python3 /Users/sekhmet/Programs/PyDPainter-1.2.2/PyDPainter.py"  
	
end run
```
It works for me on High Sierra & Catalina, but I haven’t tried it on other Mac OS’s.

**Comment by PerspexSphinx** [comment]
If you are using Thonny… 

```
(* My Brilliant AppleScript to run PyDPainter
from Thonny…
Note: Path to Python should be O.K. but…  
Change your Path to PyDPainter.py *)

on run
	activate
	do shell script "/Applications/Thonny.app/Contents/Frameworks/Python.framework/Versions/3.10/bin/Python3 /Users/charybdis/Programs/PyDPainter-1.2.2/PyDPainter.py"
end run
```

**Comment by mriale** [comment]
Sorry about my Mac unfamiliarity, but can you double click on the PyDPainter.py file to launch it? Can you drag it into the dock?

If not, a shell script could work too. Save a file called "PyDPainter.sh" in the same directory and put this in it:
`python PyDPainter.py`

Then double-click the shell script or drag it onto the dock. You might have to make it executable before running it.

Maybe these things don't work on the Mac. I only have Linux and Windows, so I can't try it.

**Reply by PerspexSphinx** [reply]
No, I can’t just double click on PyDPainter.py to launch it, maybe that’s because I’m using the python that comes with Thonny 4.1.4, but a shell script works…

PyDPainter.sh
```
/Applications/Thonny.app/Contents/Frameworks/Python.framework/Versions/3.10/bin/Python3 /Users/charybdis/Programs/PyDPainter-1.2.2/PyDPainter.py
```
but it leaves the terminal open & also,  I can’t seem to put it in the dock?

Whereas an AppleScript saved as an application (hold down the Option key in the File menu of the AppleScript editor to get save as…) then change the file format to “application”.
The resulting “app” can be placed in the dock and even given a distinct icon
```
(* My AppleScript code…
to run Mark Riale's Brilliant PyDPainter
Note:In this case for python3 from Thonny
Change Path to Python & Path to PyDPainter.py, appropriately *)

on run
	activate
	do shell script "/Applications/Thonny.app/Contents/Frameworks/Python.framework/Versions/3.10/bin/Python3 /Users/charybdis/Programs/PyDPainter-1.2.2/PyDPainter.py"
end run
```

Thanks, for your **Wonderful** Re-Animated DPaint (I’m looking forward to it's Future and Animation capabilities)

**Comment by MetinSeven** [comment]
I'm reading this with great interest.

I'll be returning to macOS this year, and would love to keep using PyDPainter, but I'm not familiar with scripting or compiling. Is there a macOS executable available for download somewhere?

Thanks.

**Reply by mriale** [reply]
I haven't built a Mac version. Here's why:
- It costs $100 a year to be a Mac developer to be able to publish apps
- It requires a Mac computer to run PyInstaller build the apps, and the cheapest Mac mini is $600

I guess I could charge money on the app store to recover these costs. PyDPainter currently has 2,473 Linux install from Flathub, so at $1 each I would still have a profit.

Do you think Mac people would pay money for a free app? How much do programs like this go for?

**Comment by MetinSeven** [comment]
Hi Mark,

I expect macOS users to be willing to pay for a solid macOS version of PyDPainter, as Mac users are generally speaking more used to paying for their software than Windows or Linux users. The same goes for iOS users versus Android users.

If PyDPainter for macOS would cost something like $ 5, I would not hesitate to buy it. And I would let people know of its availability via social media and forums.

Thanks for possibly considering a macOS version. The M4 Mac Mini is a gem of a machine for a very sharp price. You won't regret buying one.

**Comment by pauljsymonds** [comment]
This works fine on Mac Silicon, just need to create the Environment and run it.
